src/libcore/tests/ptr.rs

```diff
@@ -164,7 +163,7 @@ fn test_as_mut() {
         // Pointers to unsized types -- slices
         let s: &mut [u8] = &mut [1, 2, 3];
         let ms: *mut [u8] = s;
-        assert_eq!(ms.as_mut(), Some(s));
+        assert_eq!(ms.as_mut().unwrap() as *mut _, s as *mut _);
```

Comment:
```suggestion
        assert_eq!(ms.as_mut(), Some(&mut [1, 2, 3]));
```

I think the tests is intended to compare *contents* and not the pointer value.

Reply:
Oh. Well that's fine for me too.